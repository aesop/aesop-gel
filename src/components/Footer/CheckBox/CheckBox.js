import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './CheckBox.module.css';

const CheckBox = ({ className, content, disabled, onChange }) => {
  const baseClassSet = cx(styles.base, className);
  const checkboxClassSet = cx(styles.checkBox, className);
  const contentClassSet = cx(styles.content, className);

  return (
    <div className={baseClassSet}>
      <input
        className={checkboxClassSet}
        disabled={disabled}
        onChange={onChange}
        type="checkbox"
      />
      <div className={contentClassSet}>{content}</div>
    </div>
  );
};

CheckBox.propTypes = {
  className: PropTypes.string,
  content: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
};

CheckBox.defaultProps = {
  className: undefined,
  content: undefined,
  disabled: false,
  onChange: undefined,
};

export default CheckBox;
