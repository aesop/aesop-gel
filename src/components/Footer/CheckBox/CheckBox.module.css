@import 'styles/index';

.base {
  display: flex;
  margin-top: 18px;
}

.checkBox {
  display: inline-block;
  width: 13px;
  height: 13px;
  flex: 0 0 13px;
  border: 1.5px solid var(--color-white);
  margin: 7.5px 0 0 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-clip: content-box;
  background-color: transparent;
  color: var(--color-white);
  font-size: rem(14px);
  line-height: 2.14;

  &:checked::before {
    position: relative;
    top: -7px;
    content: '\2713';
  }

  &:focus {
    outline: none !important;
  }

  &:hover {
    cursor: pointer;
  }
}

.content {
  margin-left: 10px;
  color: var(--color-white);
  font-size: rem(14px);
  line-height: 2.14;
}
