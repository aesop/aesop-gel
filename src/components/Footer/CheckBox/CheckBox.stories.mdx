import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import Podium from '~/components/Podium';
import CheckBox from './CheckBox';

<Meta title="Components/CheckBox" component={CheckBox} />

# CheckBox

<Preview>
  <Story name="Base component">
    <Podium horizontalPadding="small" verticalPadding="small">
      <p>Set the background to dark</p>
      <CheckBox
        content='Subscribe to communications about Aesop products, services, stores, events and matters of cultural interest.'
      />
    </Podium>
  </Story>
</Preview>

## API

<Props of={CheckBox} />
