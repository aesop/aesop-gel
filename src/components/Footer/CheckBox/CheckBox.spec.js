import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CheckBox from './CheckBox';

configure({ adapter: new Adapter() });

describe('<CheckBox />', () => {
  it('should be defined', () => {
    expect(CheckBox).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <CheckBox
          content="Subscribe to communications about Aesop products, services, stores, events and matters of cultural interest."
          name="test-checkbox"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
