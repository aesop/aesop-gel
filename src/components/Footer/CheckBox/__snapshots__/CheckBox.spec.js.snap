// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> renders base component correctly 1`] = `
<label
  className="base"
  htmlFor="test-checkbox"
>
  <input
    className="checkBox"
    disabled={false}
    id="test-checkbox"
    type="checkbox"
  />
  <span
    className="content"
  >
    Subscribe to communications about Aesop products, services, stores, events and matters of cultural interest.
  </span>
</label>
`;
